src/ForceHighPoly.Core/ForceHighPoly.Hooks.cs

```diff
@@ -1,36 +1,224 @@
+                    for (var j = 0; j < clothParts.Length; j++)
                     {
-#if KKS
-                        if (assetName != "p_cf_body_00_hit_low") //Loading the high poly version of this breaks special H modes
+                        var category = 105;
+                        switch (j)
+                        {
+                            case 0:
+                                category = (__instance.sex != 0 || exTypeExists && exType.GetValue<int>() != 1) ? 105 : 503;
+                                break;
+                            case 7:
+                            case 8:
+                                category = ((__instance.sex != 0 || exTypeExists && exType.GetValue<int>() != 1) ? 112 : 504) - j;
+                                break;
+                            default:
+                                break;
+                        }
+                        category += j;
+                        var currentcloth = clothParts[j];
+                        var id = currentcloth.id;
+                        var work = __instance.lstCtrl.GetCategoryInfo((ChaListDefine.CategoryNo)category);
+                        if (!work.TryGetValue(id, out var lib))
+                        {
+                            continue;
+                        }
+                        else if (category == 105 || category == 107)
+                        {
+                            var infoInt = lib.GetInfoInt(ChaListDefine.KeyType.Sex);
+                            if (__instance.sex == 0 && infoInt == 3 || __instance.sex == 1 && infoInt == 2)
+                            {
+                                if (id != 0)
+                                {
+                                    work.TryGetValue(0, out lib);
+                                }
+                                if (lib == null)
+                                {
+                                    continue;
+                                }
+                            }
+                        }
+                        var highAssetName = lib.GetInfo(ChaListDefine.KeyType.MainData);
+                        if (string.Empty == highAssetName)
+                        {
+                            continue;
+                        }
+                        var manifestName = lib.GetInfo(ChaListDefine.KeyType.MainManifest);
+                        var assetBundleName = lib.GetInfo(ChaListDefine.KeyType.MainAB);
+#if KK
+                        Singleton<Manager.Character>.Instance.AddLoadAssetBundle(assetBundleName, manifestName);
+#elif KKS
+                        Manager.Character.AddLoadAssetBundle(assetBundleName, manifestName);
 #endif
+                        if (!CommonLib.LoadAsset<UnityEngine.GameObject>(assetBundleName, lib.GetInfo(ChaListDefine.KeyType.MainData) + "_low", false, manifestName) && CommonLib.LoadAsset<UnityEngine.GameObject>(assetBundleName, highAssetName, false, manifestName))
```

Comment:
Does this get reused? It seems like this would slow down the load process, and if only one piece of clothing is missing a low poly version then others will get their low and high poly versions loaded. I think there's a method for checking if an asset exists, maybe that can be used?

Reply:
This is postfixed to ChaControl initialization so it's before the character gets loaded and doesn't get caught up having to interact with any Async patches.

Reply:
Tried looking into this, but looking for the _low asset's existence seems to be approximately equal amount of work and needing new code